src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -280,13 +282,13 @@ public BaseResponseBean advanceSessionWithSelections(MenuSession menuSession,
         }
     }
 
-    private NotificationMessage handleQueryScreen(Screen nextScreen, MenuSession menuSession,
-                                                  boolean forceManualAction, QueryData queryData)
+    private NotificationMessage handleQueryScreen(Screen nextScreen, MenuSession menuSession, QueryData queryData,
+                                                  boolean forceSearch, boolean disallowSearch)
             throws CommCareSessionException {
         if (nextScreen instanceof FormplayerQueryScreen) {
             FormplayerQueryScreen formplayerQueryScreen = ((FormplayerQueryScreen)nextScreen);
             formplayerQueryScreen.refreshItemSetChoices();
-            boolean autoSearch = formplayerQueryScreen.doDefaultSearch() && !forceManualAction;
+            boolean autoSearch = forceSearch || (formplayerQueryScreen.doDefaultSearch() && !disallowSearch);
```

Comment:
@orangejenny What does `forceSearch` mean to solve for ? Is there an issue currently without adding this.

Reply:
This is an issue with the replay behavior. What happens is you have a parent-child workflow where the parent did an ES search (because it was either search-first or skip to ES results), and then later in the session, when the parent case is being re-selected, formplayer needs to know to run the search again. Otherwise it gets "stuck" on the formplayer query screen.

Reply:
I see, although these changes doesn't seem harmful I am not so sure if these should be necessary mainly because all the new parent child tests passes without these changes.  For Skip to ES Results since the default search is on `autoSearch` should evaluate to `true` always and for the Search-first, the `execute` param take care of moving forward the search. Is it the case that sometimes `execute` is false for the required search while replaying the session ?

Reply:
That is a good point about the tests. I didn't get to this today but will take a closer look.

Reply:
Looked at this more, and I do think this needs to be fixed on the formplayer side. What I'd forgotten in the earlier reply is that this fixes the scenario where you're using "Search Again" for the child menu. I don't think we had any tests where `forceManualAction` is true. Added one in ed782542 that fails before this commit and passes afterward.

This code is basically accounting for the fact that `forceManualSearch` is set on the entire request, it isn't specific to the menu. So when `forceManualSearch` is set because you're searching again on the child menu, that was also being applied to the parent and overriding the parent's "skip to ES results" workflow. I took a look at making `forceManualSearch` a part of `queryData` instead, but had less luck with that.

Reply:
Ahh I see, this makes much more sense now. Thanks for taking another paas at it.